llmfoundry/models/mpt/configuration_mpt.py

```diff
@@ -222,6 +223,20 @@ def _validate_config(self) -> None:
                 'attn_impl'] not in ['torch', 'triton']:
             raise NotImplementedError(
                 'prefix_lm only implemented with torch and triton attention.')
+
+        if self.attn_config['attn_impl'] == 'flash' and is_flash_v1_installed():
+            warnings.warn(
+                DeprecationWarning(
+                    'Support for Flash Attention v1 is deprecated. Please upgrade to Flash Attention v2.4.2.'
+                ))
+
+        if self.attn_config[
```

Comment:
does prefix lm require triton?

Reply:
Yes, we need triton for prefix lm because flash attention does not support prefix lm right now.

Reply:
😢

Reply:
😢